src/Native/Time.js

```diff
@@ -10,39 +10,46 @@ Elm.Native.Time.make = function(elm) {
   var Maybe = Elm.Maybe.make(elm);
   var Utils = Elm.Native.Utils.make(elm);
 
+  function fst(pair) {
+      return pair._0;
+  }
+
+  function snd(pair) {
+      return pair._1;
+  }
+
   function fpsWhen(desiredFPS, isOn) {
     var msPerFrame = 1000 / desiredFPS;
-    var prev = elm.timer.now(), curr = prev, diff = 0, wasOn = true;
-    var ticker = NS.input(diff);
-    function tick(zero) {
-      return function() {
-        curr = elm.timer.now();
-        diff = zero ? 0 : curr - prev;
-        if (prev > curr) {
-          diff = 0;
-        }
-        prev = curr;
-        elm.notify(ticker.id, diff);
-      };
-    }
+    var prev, curr, diff, wasOn = true;
+    var zero = true;
+    var ticker = NS.input(zero);
     var timeoutID = 0;
     function f(isOn, t) {
       if (isOn) {
-        timeoutID = elm.setTimeout(tick(!wasOn && isOn), msPerFrame);
+        timeoutID = elm.setTimeout( function() { elm.notify(ticker.id, !wasOn && isOn); } , msPerFrame);
       } else if (wasOn) {
         clearTimeout(timeoutID);
       }
       wasOn = isOn;
       return t;
     }
-    return A3( Signal.map2, F2(f), isOn, ticker );
+    function g(event, old) {
+      prev = snd(old);
+      curr = fst(event);
+      zero = snd(event);
+      diff = zero ? 0 : curr - prev;
+      return Utils.Tuple2(diff, curr);
```

Comment:
Yes, I had not made them local purely because I wanted to create as small as possible a conceptual diff against the old version of this function, where these variables were declared at the top.

But yes, I'll refactor now as you suggest.